metricbeat/module/ceph/mgr_cluster_health/mgr_cluster_health.go

```diff
@@ -0,0 +1,97 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package mgr_cluster_health
+
+import (
+	"fmt"
+
+	"github.com/elastic/beats/metricbeat/helper"
+	"github.com/elastic/beats/metricbeat/mb"
+	"github.com/elastic/beats/metricbeat/mb/parse"
+)
+
+const (
+	defaultScheme      = "https"
+	defaultPath        = "/request"
+	defaultQueryParams = "wait=1"
+
+	cephStatusPrefix         = "status"
+	cephTimeSyncStatusPrefix = "time-sync-status"
+)
+
+var (
+	hostParser = parse.URLHostParserBuilder{
+		DefaultScheme: defaultScheme,
+		DefaultPath:   defaultPath,
+		QueryParams:   defaultQueryParams,
+	}.Build()
+)
+
+func init() {
+	mb.Registry.MustAddMetricSet("ceph", "mgr_cluster_health", New,
+		mb.WithHostParser(hostParser),
+	)
+}
+
+type MetricSet struct {
+	mb.BaseMetricSet
+	*helper.HTTP
+}
+
+func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
+	http, err := helper.NewHTTP(base)
+	if err != nil {
+		return nil, err
+	}
+	http.SetMethod("POST")
+	http.SetHeader("Content-Type", "application/json")
+	http.SetHeader("Accept", "application/json")
+
+	return &MetricSet{
+		base,
+		http,
+	}, nil
```

Comment:
Nit. Constructor seems to be mostly the same for these metricsets, we could consider creating a common one, as is done for example in the rabbitmq module (see https://github.com/elastic/beats/blob/master/metricbeat/module/rabbitmq/metricset.go#L34).

Reply:
Fixed, extracted to a common package.